Add license field descriptions table (#582)
csg-org/CompactConnect: backend/compact-connect/docs/README.md

### Generating a CSV export of your license data

Export your license data to a CSV file, formatted as follows:
   - The file must be a utf-8 encoded text format
   - The first line must be a header with column names exactly matching the field names listed in [the table below](#field-descriptions).
   - All subsequent lines must be individual licenses.
   - At least all required fields must be present as a column and required fields cannot be empty in any row.
   - Any optional fields may also be included. Optional fields can be left empty in some rows.
   - Order of columns does not matter.
   - String lengths are enforced - exceeding them will cause validation errors

>    - Enum values must match the allowed values exactly, including spacing and capitalization

Reviewer: Not sure `enum` is a widely used term. Perhaps–

```suggestion
   - Some fields have a set list of allowed values. For those fields, make sure to enter the value exactly, including spacing and capitalization.
```

Author: Ok, these all make sense - I'll all these revisions into a single commit 👍

#### Field Descriptions

The following table describes all available fields for the license CSV file. Required fields are marked with an asterisk (*).

| Field Name | Description | Format | Example |
|------------|-------------|---------|---------|
| dateOfBirth* | Provider's date of birth | YYYY-MM-DD | 1980-01-31 |
| dateOfExpiration* | License expiration date | YYYY-MM-DD | 2025-12-31 |
| dateOfIssuance* | Date when license was originally issued | YYYY-MM-DD | 2020-01-01 |
| dateOfRenewal* | Most recent license renewal date | YYYY-MM-DD | 2023-01-01 |
| familyName* | Provider's family/last name | String (max 100 chars) | Smith |
| givenName* | Provider's given/first name | String (max 100 chars) | John |
| homeAddressCity* | City of provider's home address | String (max 100 chars) | Springfield |

> | homeAddressPostalCode* | Postal/ZIP code of provider's home address | String (5-7 chars) | 12345 |

Reviewer: Assuming only ZIP codes are allowed, shouldn't this be either 5 digits, or 9 digits? When would there be a 6 or 7-digit zip?

Author: I'm not assuming only ZIP codes. We actually recently had a conversation asking whether we should also include a field for country code in the off chance that somebody has reason to provide an international address. We haven't done that, but the Postal code at least is long enough to support a postal code from the US, Canada, or Mexico.

Reviewer: Got it– thanks!

| homeAddressState* | State/province of provider's home address | String (max 100 chars) | IL |
| homeAddressStreet1* | First line of provider's street address | String (max 100 chars) | 123 Main St |

> | licenseType* | Type of professional license. Types you provide must be associated with the compact you are uploading for. | Enum: `audiologist`, `speech-language pathologist`, `occupational therapist`, `occupational therapy assistant`, `licensed professional counselor` | occupational therapist |

Reviewer: ```suggestion
| licenseType* | Type of professional license. Types you provide must be associated with the compact you are uploading for. | One of: `audiologist`, `speech-language pathologist`, `occupational therapist`, `occupational therapy assistant`, `licensed professional counselor` | occupational therapist |
```

| ssn* | Social Security Number | Format: XXX-XX-XXXX | 123-45-6789 |

> | status* | Current status of the licensee. "active" means they will be allowed to purchase privileges that expire on the same day as their license. "inactive" means they are not eligible to purchase privileges. | Enum: `active`, `inactive` | active |

Reviewer: ```suggestion
| status* | Current status of the licensee. "active" means they will be allowed to purchase privileges that expire on the same day as their license. "inactive" means they are not eligible to purchase privileges. | One of: `active`, `inactive` | active |
```

| emailAddress | Provider's email address (optional) | Email (max 100 chars) | john.smith@example.com |
| homeAddressStreet2 | Second line of provider's street address (optional) | String (max 100 chars) | Suite 100 |
| licenseNumber | License number (optional) | String (max 100 chars) | OT12345 |
| middleName | Provider's middle name (optional) | String (max 100 chars) | Robert |

> | militaryWaiver | Whether provider has military waiver (optional) | Boolean | true |

Reviewer: ```suggestion
| militaryWaiver | Whether provider has military waiver (optional) | One of: `true`, `false` | true |
```

| npi | National Provider Identifier (optional) | 10-digit number | 1234567890 |

> | phoneNumber | Provider's phone number (optional) | [ITU-T E.164 format](https://www.itu.int/rec/T-REC-E.164-201011-I/en) | +12025550123 |

Reviewer: That specification is a beast. Can we include a tldr; summary of this format here? e.g., if this is true, maybe something like–

```suggestion
| phoneNumber | Provider's phone number (optional) | [ITU-T E.164 format](https://www.itu.int/rec/T-REC-E.164-201011-I/en) (must include country code, no spaces or dashes) | +12025550123 |
```

> | suffix | Provider's name suffix (optional) | String (max 100 chars) | Jr. |

Reviewer 2: Do we have an example csv file somewhere committed to the project that teams could copy/paste? That might be useful.
